testutil/network/gatewaynet/network.go

```diff
@@ -0,0 +1,187 @@
+package gatewaynet
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	testcli "github.com/cosmos/cosmos-sdk/testutil/cli"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/testutil/network"
+	"github.com/pokt-network/poktroll/testutil/network/basenet"
+	"github.com/pokt-network/poktroll/testutil/testkeyring"
+	"github.com/pokt-network/poktroll/x/application/client/cli"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+var _ network.InMemoryNetwork = (*inMemoryNetworkWithGateways)(nil)
+
+// inMemoryNetworkWithGateways is an implementation of the InMemoryNetwork interface.
+type inMemoryNetworkWithGateways struct {
+	basenet.BaseInMemoryNetwork
+}
+
+// DefaultInMemoryNetworkConfig returns the default in-memory network configuration.
+// This configuration should sufficiently populate on-chain objects to support reasonable
+// coverage around most session-oriented scenarios.
+func DefaultInMemoryNetworkConfig(t *testing.T) *network.InMemoryNetworkConfig {
+	t.Helper()
+
+	return &network.InMemoryNetworkConfig{
+		NumGateways:             5,
+		NumSuppliers:            2,
+		AppSupplierPairingRatio: 1,
```

Comment:
Can we add a comment explaining this field

Reply:
It's documented in the declaration: https://github.com/pokt-network/poktroll/blob/main/testutil/network/types.go#L46